paasta_tools/tron_tools.py

```diff
@@ -912,6 +951,9 @@ def format_tron_action_dict(action_config: TronActionConfig):
         result["executor"] = EXECUTOR_NAME_TO_TRON_EXECUTOR_TYPE.get(
             executor, "kubernetes"
         )
+        if executor == "spark":
+            # inject spark configs to the original spark-submit command
+            action_config.action_spark_config = action_config.build_spark_config()
```

Comment:
just curious: how come we had to move this up?

Reply:
`get_env` was called above this, before the spark config had been built and set, resulting in incorrect values in env variables